### ISSUE TITLE
Timestamps can be altered using the filesystem exchange

### ISSUE THREAD

**Issue by rkdarst** [jupyter/nbgrader#1104]
Related to #1097 (user impersonation using the exchange), I think users can adjust files or timestamps after submission when using the filesystem exchange.  My understanding is that the exchange only uses timestamp.txt (for a good reason in most cases), but because these files are accessible to students there's not much to do.

Things are a bit obfuscated: now that we have random strings in submissions, the submission log file doesn't say what they are.  But if assignment is submitted from the command line, it shows the random string and enough to find the submission directory.  Then timestamp.txt can presumably be altered or the submitted `.ipynb` file can be changed.  My workaround is telling instructors to not rely on that, until you collect you can't know anything.

One could use filesystem timestamps (related to #1097), but that will assume there are no copies (or copies careful to preserve timestapms) before collection...

This is perhaps minor, but thought I would document it.

I could also be wrong and it uses filesystem times somehow, in which case this can be closed - I only checked a while ago.

**Comment by jhamrick** [comment]
Yes, I think that's correct. I'm not sure it's worth trying to fix it, but I do think it is worthwhile to document it at least. Perhaps once we create the exchange API we can create a docs page discussing how to create your own exchange if you want to do that, and the limitations of nbgrader's current filesystem exchange.

**Comment by rkdarst** [comment]
Oh, I agree it's not worth fixing beyond documenting and considering for future exchange design.  Fixing here is really hard, fixing in a proper exchange service is easy.  The exchange is great for what it is.  (same for #1097).  Thank you for all your work on nbgrader, it has helped a lot of people here greatly!  I don't mean anything bad about it.

**Comment by jhamrick** [comment]
Oh, I didn't think you did! Apologies if my response came off a bit terse, I'm currently reading through loads of issues trying to decide what should go into the next release of 0.6.0 vs 0.7.0 and my eyes are starting to glaze over a bit 🙂

Thanks for reporting all these issues (and for the PRS!), I think it's really important to document all this stuff and it's easy for smaller things like this to slip through the cracks.
